From b6924f0b1bc22b84c220ebeae4de8e0d145076f2 Mon Sep 17 00:00:00 2001
From: Suren Baghdasaryan <surenb@google.com>
Date: Tue, 23 Aug 2022 13:25:56 -0700
Subject: [PATCH 1/5] lmkd: Remove process record after it is killed by lmkd
 watchdog

After lmkd watchdog kills a process, its record should be removed.

Bug: 243567425
Signed-off-by: Suren Baghdasaryan <surenb@google.com>
Change-Id: I70bb2a432df8088ebc9865fbc36b065738248d19
---
 lmkd.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lmkd.cpp b/lmkd.cpp
index a87bb7a..a26d0d8 100644
--- a/lmkd.cpp
+++ b/lmkd.cpp
@@ -2153,6 +2153,8 @@ static void watchdog_callback() {
         if (reaper.kill({ target.pidfd, target.pid, target.uid }, true) == 0) {
             ALOGW("lmkd watchdog killed process %d, oom_score_adj %d", target.pid, oom_score);
             killinfo_log(&target, 0, 0, 0, NULL, NULL, NULL, NULL, NULL);
+            // WARNING: do not use target after pid_remove()
+            pid_remove(target.pid);
             break;
         }
         prev_pid = target.pid;
-- 
2.32.0

